### PR TITLE
Lookupname eq varname

### PR DESCRIPTION
**Pull request by PatrickRWright**
It turns out that there is a problem when the name of a lookuptable (secuTrial) is equal to the name of the variable its being used in. If this is the case then the meta variables are not factorized properly.
It triggers this warning:
`warning(paste("Unexpected: Not all levels converted for", name, "in form", form))`

The fix is removing the `nrow(lookup) == 0` condition. I spent quite some time investigating why I may have put it there in the first place, and as of now I think there is no need. Next to the tests in the package which all still pass without any changes I also made several local checks with private datasets (7 different secuTrial exports including lookuptable data structures). I loaded them before removing the condition and after removing the condition and checked the outputs with `all.equal()`. For all exports except the one that triggered the issue in the first place `all.equal()` returned `TRUE`.

In summary I think this condition can be safely removed because it serves no purpose and creates problems in specific scenarios.

### CODE REVIEW

**Comment by aghaynes** [comment]
even though it's small, it would be good to have it documented in NEWS for the sake of completeness... just put it as a bug fix and reference this PR
